ubi_manifest/worker/tasks/depsolver/notifications.py

```diff
@@ -0,0 +1,126 @@
+"""
+Notification sending module
+"""
+
+import smtplib
+from datetime import datetime
+from typing import List
+
+from .models import NotifierConfig, NotifierItem
+
+
+class Notifier:
+    """
+    Module for sending email notifications.
+    """
+
+    def __init__(self, config: NotifierConfig) -> None:
+        self.smtp_host = config.smtp_host
+        self.smtp_port = config.smtp_port
+        self.to_addr = config.to_addr
+        self.from_addr = config.from_addr
+
+    def _prepare_header(self, ubi_repo_ids):
+        header = [f"From: {self.from_addr}"]
+        if len(ubi_repo_ids) == 1:
+            header.append(
+                f"Subject: Failed depsolving UBI repository {ubi_repo_ids[0]} "
+            )
+        else:
+            repo_id_string = " and ".join(ubi_repo_ids)
+            # don't make the subject too long
+            if len(repo_id_string) > 60:
+                repo_id_string = ubi_repo_ids[0] + "..."
+            header.append(
+                f"Subject: Failed depsolving UBI repositories {repo_id_string}"
+            )
+        header.append(f"To: {self.to_addr}")
+        header.append("")
+        return header
+
+    def send_notification(
+        self,
+        ubi_repo_ids: List[str],
+        rpm_issues: NotifierItem,
+        debug_issues: NotifierItem,
+    ) -> None:
+        """
+        Prepare a notification email and send it.
+        """
+        body = _prepare_body(rpm_issues, debug_issues)
+        # Avoid sending empty message
+        if body:
+            msg = self._prepare_header(ubi_repo_ids)
+
+            msg.append("While resolving dependencies for the repositories")
+            for repo in ubi_repo_ids:
+                msg.append(f"     {repo}")
+            msg.append("we have encountered these issues:")
+            msg.append("")
+
+            msg.extend(body)
+
+            msg.append(
+                f"The issues were reported at: {datetime.utcnow().isoformat()[:-7]} UTC"
+            )
+            message = "\r\n".join(msg)
+            # Send message
+            server = smtplib.SMTP(self.smtp_host, self.smtp_port)
+            server.sendmail(self.from_addr, self.to_addr, message)
```

Comment:
I could put this into try block to prevent the workers from crashing if the smtp server is temporarily unavailable. (Or I could put it into a thread and retry sending a few times if it fails.)